src/main/kotlin/com/cognifide/gradle/aem/common/instance/LocalInstanceManager.kt

```diff
@@ -351,4 +354,75 @@ class LocalInstanceManager(private val aem: AemExtension) : Serializable {
 
         return upInstances
     }
+
+    /**
+     * Pre-conditionally check if 'java' is available in shell scripts.
+     *
+     * Gradle is intelligently looking by its own for installed Java, but AEM control scripts
+     * are just requiring 'java' command available in 'PATH' environment variable.
+     */
+    @Suppress("TooGenericExceptionCaught", "MagicNumber")
+    fun examineJavaAvailable() {
+        try {
+            logger.info("Examining Java properly installed")
+            val result = ProcBuilder("java", "-version")
+                    .withWorkingDirectory(rootDir.get().asFile)
+                    .withTimeoutMillis(5000)
+                    .withExpectedExitStatuses(0)
+                    .run()
+            logger.info("Installed Java:\n${result.outputString.ifBlank { result.errorString }}")
+        } catch (e: Exception) {
+            throw LocalInstanceException("Local instances support requires Java properly installed! Cause: '${e.message}'\n" +
+                    "Ensure having directory with 'java' executable listed in 'PATH' environment variable.", e)
+        }
+    }
+
+    /**
+     * Defines compatibility related to AEM versions and Java versions.
+     *
+     * AEM version is definable as range inclusive at a start, exclusive at an end.
+     * Java Version is definable as list of supported versions pipe delimited.
+     */
+    val javaCompatibility = aem.obj.map<String, String> {
```

Comment:
pipes are here due to compatibility with delimiters used by `PropertyParser.map()`